### PR TITLE
add docstrings for front, tail

### PR DESCRIPTION
**Pull request by jw3126**
These two functions are handy and I guess also unlikely to experience breaking changes. Maybe also export them?

### CODE REVIEW

**Comment by mbauman** [comment]
> unlikely to experience breaking changes

Somewhat amusingly, this PR appears to be such a "breaking" change.  That said, I'd support unifying the error types between these two functions — you'll just have to fixup the tests.

I'd support exporting `tail` and `front`, but IIRC previous discussions stalled on the fact that it's not a very generic operation.  Making indexing by constant ranges fast is another alternative: #31138.

**Comment by jw3126** [comment]
The travis fail looks unrelated to the PR.

**Comment by jw3126** [comment]
bump

**Comment by jw3126** [comment]
Merge?

File: base/essentials.jl [hunk 1]
```diff
@@ -170,7 +170,21 @@ macro eval(mod, ex)
 end
 
 argtail(x, rest...) = rest
+
+"""
+    tail(x::Tuple)::Tuple
+
+Return a `Tuple` consisting of all but the first component of `x`.
+```jldoctest
```
Comment (fredrikekre) [inline base/essentials.jl]:
```suggestion

# Examples
```jldoctest
```

File: base/essentials.jl [hunk 2]
```diff
@@ -170,7 +170,21 @@ macro eval(mod, ex)
...
+julia> Base.tail((1,2,3))
+(2, 3)
+
+julia> Base.tail(())
+ERROR: ArgumentError: Cannot call tail on an empty tuple.
+```
+"""
 tail(x::Tuple) = argtail(x...)
+tail(::Tuple{}) = throw(ArgumentError("Cannot call tail on an empty tuple."))
```
Comment (fredrikekre) [inline base/essentials.jl]: Add a test for this?

File: base/tuple.jl [hunk 3]
```diff
@@ -107,11 +107,23 @@ safe_tail(t::Tuple{}) = ()
 
 # front (the converse of tail: it skips the last entry)
 
+"""
+    front(x::Tuple)::Tuple
+
+Return a `Tuple` consisting of all but the last component of `x`.
+```jldoctest
```
Comment (fredrikekre) [inline base/tuple.jl]:
```suggestion

# Examples
```jldoctest
```

File: base/essentials.jl (hunk 2 above)
Comment (jw3126) [reply base/essentials.jl]: Note there was already a doctest, but I also added a test.

**Comment by jw3126** [comment]
@fredrikekre LGTY?

**Review by fredrikekre — APPROVED** [review]
There was a failure during doc build but I believe that issue has been fixed. Restarted just in case.

**Pull request merged**